packages/lab/src/pill-next/PillNext.tsx

```diff
@@ -34,34 +27,52 @@ export const PillNext = forwardRef<HTMLButtonElement, PillNextProps>(
       css: pillCss,
       window: targetWindow,
     });
+    const density = useDensity();
+    const closeIcon = density === "high" ? <CloseSmallIcon /> : <CloseIcon />;
     const { buttonProps, active } = useButton<HTMLButtonElement>({
       disabled,
       ...restProps,
     });
     // we do not want to spread tab index in this case because the button element
     // does not require tabindex="0" attribute
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
     const { tabIndex, ...restButtonProps } = buttonProps;
 
+    const handleClose = (event: MouseEvent<HTMLButtonElement>) => {
```

Comment:
This is 100% nit picking but handleClose isn't really needed here. onClose can be immediately passed as a prop to <Buttton>